wallet/src/extkey.rs

```diff
@@ -202,43 +205,30 @@ impl ExtendedKey {
 		Ok(ext_key)
 	}
 
-	/// Return the identifier of the key, which is the
-	/// Hash160 of the private key
+	/// Return the identifier of the key
+	/// which is the blake2b hash (20 bit digest)
 	pub fn identifier(&self) -> Identifier {
-		let mut sha = Sha256::new();
-		sha.input(&self.key[..]);
-
-		let mut shres = [0; 32];
-		sha.result(&mut shres);
-
-		let mut ripe = Ripemd160::new();
-		ripe.input(&shres[..]);
-
-		let mut identifier = [0; 20];
-		ripe.result(&mut identifier);
-		Identifier::from_bytes(&identifier)
+		let identifier = blake2b(20, &[], &self.key[..]);
+		Identifier::from_bytes(&identifier.as_bytes())
 	}
 
 	/// Derive an extended key from an extended key
 	pub fn derive(&self, secp: &Secp256k1, n: u32) -> Result<ExtendedKey, Error> {
-		let mut hmac = Hmac::new(Sha512::new(), &self.chaincode[..]);
 		let mut n_bytes: [u8; 4] = [0; 4];
 		BigEndian::write_u32(&mut n_bytes, n);
+		let mut seed = self.key[..].to_vec();
+		seed.extend_from_slice(&n_bytes);
 
-		hmac.input(&self.key[..]);
-		hmac.input(&n_bytes[..]);
-
-		let mut derived = [0; 64];
-		hmac.raw_result(&mut derived);
+		let derived = blake2b(64, &self.chaincode[..], &seed[..]);
```

Comment:
Can someone double check the key and the msg are in the right places here in this call?

Reply:
Looks good to me